### ISSUE TITLE
Extracting of people, organizations and locations fails silently

### ISSUE THREAD

**Issue by DrWhax** [ICIJ/datashare#88]
Currently, in Datashare 1.45 (Linux) (but probably below as well, but I haven't tested) I have ~11000 documents that are indexed. When I want to extract people, organizations and locations. I see the following in my terminal output:

```
2019-03-04 13:03:44,715 [Worker: RequestDispatcher: Thread-29] INFO  AbstractModels - synchronize models is set to true
2019-03-04 13:03:44,715 [Worker: RequestDispatcher: Thread-29] INFO  TaskResource - waiting for NlpApp CORENLP to listen...
2019-03-04 13:03:44,715 [pool-3-thread-1] INFO  NlpApp - running NlpApp for CORENLP pipeline with 1 thread(s)
2019-03-04 13:03:44,716 [pool-3-thread-1] INFO  NlpForwarder - waiting for messages on host [redis]
2019-03-04 13:03:44,717 [Worker: RequestDispatcher: Thread-29] INFO  TaskResource - ...CORENLP is listening
2019-03-04 13:03:44,717 [pool-3-thread-2] INFO  ResumeNlpTask - resuming NLP name finding for user local and [CORENLP]
2019-03-04 13:03:45,589 [CORENLP-0] INFO  NlpConsumer - extracting CORENLP entities for document 5c9dc3f5220ad125d3c646c98322380662bb6fabc25cc70aaf8b9d5582e74954
2019-03-04 13:03:45,590 [CORENLP-0] INFO  CorenlpPipeline - initializing CORENLP ENGLISH [TOKEN, SENTENCE, POS, LEMMA, NER]
2019-03-04 13:03:46,626 [pool-3-thread-2] INFO  ResumeNlpTask - sent EXTRACT_NLP message for 10000 files without [CORENLP] pipeline tags
2019-03-04 13:03:47,135 [pool-3-thread-1] INFO  RedisSubscriber - Shutdown called. Unsubscribe done.
2019-03-04 13:03:47,136 [pool-3-thread-1] INFO  NlpApp - forwarder exited waiting for consumer(s) to finish
2019-03-04 13:03:53,768 [Worker: RequestDispatcher: Thread-25] INFO  TaskManager - cancelling task org.icij.datashare.text.nlp.NlpApp[CORENLP]@2efc96cf
```
It seems it doesn't really like a lot of documents so it fails silently, there's no indication in the web client that it has installed. 

When I subsequently want to kill the job I get this error message:

```
2019-03-04 13:03:53,769 [pool-3-thread-1] ERROR NlpApp - error running NlpApp
java.lang.InterruptedException: null
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Object.java:502)
        at org.icij.datashare.text.nlp.NlpApp.waitForQueueToBeEmpty(NlpApp.java:101)
        at org.icij.datashare.text.nlp.NlpApp.shutdown(NlpApp.java:84)
        at org.icij.datashare.text.nlp.NlpApp.run(NlpApp.java:76)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
2019-03-04 13:03:53,769 [pool-3-thread-1] INFO  NlpApp - exiting run
```

I hope this bug report is useful. Let me know if I can do anything or provide you with more data.

**Comment by DrWhax** [comment]
Even though all those jobs are allegedly cancelled, in the output I see that CoreNlp is actually running and extracting. Which is something i'm happy about! But i'm not sure if this is intended behaviour after the previous messages.

This processing is not represented in the web interface.

**Comment by Soliine** [comment]
Thanks a lot for this issue. As we have team members who are off this week, please do not worry if you have no answer about this this week. We will look into it next week.

**Comment by DrWhax** [comment]
Cheers! :) :+1:

**Comment by bamthomas** [comment]
Hello @DrWhax thank you for your feedbacks. So yes for now we have put a limit of 10000 in the NLP extraction task, to avoid overloading the user's machine.

It is actually a bit arbitrary because what is taking most of the resources is the loading in memory of the NLP models (so if we have a corpus of 5000 docs with one language it will be less resource consuming than 5 docs with 5 different languages). 

Whatever, it should not block. What I find weird is that there is no downloading or loading model log. I will try to reproduce this on my computer. 

Maybe you could also wait for more than 6 seconds, the model was maybe loading, the english NLP model is 991MB

**Comment by bamthomas** [comment]
Hello @DrWhax. Sorry to come back this late.
This issue should have been fixed with the commit 9db6376475cf4f2bd7322f7bda38cfc335bb209c
Don't hesitate to reopen if you see this issue again.

**Comment by DrWhax** [comment]
@bamthomas much appreciated!
